runtime/j9vm/jvm.c

```diff
@@ -4542,54 +4616,64 @@ JVM_RegisterSignal(jint sigNum, void* handler)
 	memset(&oldSignalAction, 0, sizeof(struct sigaction));
 #endif /* !defined(WIN32) */
 
-	Trc_SC_RegisterSignal();
+	Trc_SC_RegisterSignal_Entry(sigNum, handler);
 
-	if (isSignalUsedByVM(sigNum)) {
-		/* Don't allow user to register a native handler since
-		 * the signal is already used by the VM.
-		 */
-	} else if (J9_ARE_NO_BITS_SET(javaVM->sigFlags, J9_SIG_XRS_SYNC)
-			&& isSignalSpecial(sigNum)) {
-		/* Don't allow user to register a native handler since
-		 * the signal is already used by the VM.
+	if (isSignalReservedByJVM(sigNum)) {
+		/* Return error if signal is reserved by the JVM. */
+		return (void *)J9_SIG_ERR;
```

Comment:
This return is missing a TraceExit

Reply:
added an `exit` label. `oldHandler` is initialized to `J9_SIG_ERR` so we can `goto exit` and return `oldHandler`.